Source/render/ShapeRenderer.swift

```diff
@@ -346,3 +246,34 @@ class ShapeRenderer: NodeRenderer {
     }
 
 }
+
+extension Stroke {
+    func strokeUsingAlphaOnly() -> Stroke? {
+        let alphaFill = fill.fillUsingAlphaOnly()
+        return Stroke(fill: alphaFill!, width: width, cap: cap, join: join, dashes: dashes, offset: offset)
+    }
+}
+
+extension Fill {
+    func fillUsingAlphaOnly() -> Fill? {
```

Comment:
Unnecessary optional return. You force the caller of this method to use `!` while it's not necessary.

Reply:
Fill is either `Color` or `Gradient`, so just cast it to `Gradient` if it's not `Color`.

Reply:
Such a solution will make it easier to extend. For example, once we'll add `Pattern` fill it will just break and stack trace show you the place where it should be fixed. Otherwise it just stop working correctly and you need to find the source of the issue yourself.